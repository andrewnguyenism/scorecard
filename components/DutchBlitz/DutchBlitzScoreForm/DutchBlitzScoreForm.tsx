import { useState } from "react";

export const DutchBlitzScoreForm = ({ editMode, submitScore }) => {
  const [dutchScore, setDutchScore] = useState(0);
  const [dutchScoreError, setDutchScoreError] = useState(false);
  const [blitzScore, setBlitzScore] = useState(10);
  const [blitzScoreError, setBlitzScoreError] = useState(false);

  return (
    <div className="rounded-2xl border-2 overflow-hidden m-4 p-6">
      <div className="flex flex-col mb-4">
        <label className="uppercase font-bold text-gray-900" htmlFor="dutch">
          Dutch Score
        </label>
        <input
          className={`border py-2 px-3 text-grey-darkest ${
            dutchScoreError && "border-red-500"
          }`}
          id="dutch"
          type="text"
          inputMode="numeric"
          pattern="[0-9]*"
          placeholder="Dutch score"
          onChange={(event) => {
            const parsedValue = Number.parseInt(event.currentTarget.value, 10);
            if (Number.isNaN(parsedValue)) {
              setDutchScoreError(true);
            } else {
              setDutchScore(parsedValue);
            }
          }}
          value={dutchScore}
        />
      </div>
      <div className="flex flex-col mb-4">
        <label className="uppercase font-bold text-gray-900" htmlFor="blitz">
          Blitz Score
        </label>
        <input
          className={`border py-2 px-3 text-grey-darkest ${
            blitzScoreError && "border-red-500"
          }`}
          id="blitz"
          type="text"
          inputMode="numeric"
          pattern="[0-9]*"
          placeholder="Blitz score"
          onChange={(event) => {
            const parsedValue = Number.parseInt(event.currentTarget.value, 10);
            if (Number.isNaN(parsedValue)) {
              setBlitzScoreError(true);
            } else {
              setBlitzScore(parsedValue);
            }
          }}
          value={blitzScore}
        />
      </div>
      <div className="flex flex-col mb-4 text-center">
        <div className="uppercase font-bold text-gray-900">Score</div>
        <div className="text-lg font-bold">
          {(dutchScore - blitzScore * 2) || 0}
        </div>
      </div>
      <div>
        <button
          className="block bg-indigo-800 hover:bg-indigo-700 text-white uppercase text-sm font-semibold mx-auto px-4 py-2 rounded-2xl"
          onClick={() => {
            if (!dutchScoreError && !blitzScoreError) {
              submitScore(dutchScore, blitzScore);
              setDutchScore(0);
              setBlitzScore(10);
            }
          }}
        >
          {editMode ? "Edit Score" : "Submit"}
        </button>
      </div>
    </div>
  );
};
