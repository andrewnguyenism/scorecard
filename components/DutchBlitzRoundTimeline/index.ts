export { DutchBlitzRoundTimeline } from './DutchBlitzRoundTimeline'

export type { DutchBlitzRound } from './DutchBlitzRoundTimeline'
