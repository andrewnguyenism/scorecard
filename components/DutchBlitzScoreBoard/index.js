export {
  DutchBlitzScoreBoardHeader,
  DutchBlitzScoreBoardPlayer,
  DutchBlitzScoreBoard,
} from "./DutchBlitzScoreBoard";
