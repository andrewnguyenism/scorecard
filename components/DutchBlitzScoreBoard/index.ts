export { DutchBlitzScoreBoard } from './DutchBlitzScoreBoard'
export { DutchBlitzScoreBoardHeader } from './DutchBlitzScoreBoardHeader'
export { DutchBlitzScoreBoardPlayer } from './DutchBlitzScoreBoardPlayer'
