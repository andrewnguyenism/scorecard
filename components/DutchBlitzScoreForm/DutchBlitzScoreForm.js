import { useState } from "react";

export const DutchBlitzScoreForm = ({ editMode, submitScore }) => {
  const [dutchScore, setDutchScore] = useState(0);
  const [dutchScoreError, setDutchScoreError] = useState(false);
  const [blitzScore, setBlitzScore] = useState(10);
  const [blitzScoreError, setBlitzScoreError] = useState(false);

  return (
    <div className="rounded-2xl border-2 overflow-hidden m-4 p-6">
      <div className="flex flex-col mb-4">
        <label className="uppercase font-bold text-gray-900" htmlFor="dutch">
          Dutch Score
        </label>
        <input
          className={`border py-2 px-3 text-grey-darkest ${
            dutchScoreError && "border-red-500"
          }`}
          id="dutch"
          type="text"
          inputMode="numeric"
          pattern="[0-9]*"
          placeholder="Dutch score"
          onChange={(event) => setDutchScore(event.currentTarget.value)}
          value={dutchScore}
        />
      </div>
      <div className="flex flex-col mb-4">
        <label className="uppercase font-bold text-gray-900" htmlFor="blitz">
          Blitz Score
        </label>
        <input
          className={`border py-2 px-3 text-grey-darkest ${
            blitzScoreError && "border-red-500"
          }`}
          id="blitz"
          type="text"
          inputMode="numeric"
          pattern="[0-9]*"
          placeholder="Blitz score"
          onChange={(event) => setBlitzScore(event.currentTarget.value)}
          value={blitzScore}
        />
      </div>
      <div className="flex flex-col mb-4 text-center">
        <div className="uppercase font-bold text-gray-900">Score</div>
        <div className="text-lg font-bold">
          {parseInt(dutchScore - blitzScore * 2) || 0}
        </div>
      </div>
      <div>
        <button
          className="block bg-indigo-800 hover:bg-indigo-700 text-white uppercase text-sm font-semibold mx-auto px-4 py-2 rounded-2xl"
          onClick={() => {
            const dutchInt = parseInt(dutchScore);
            const blitzInt = parseInt(blitzScore);

            if (Number.isNaN(dutchInt)) {
              setDutchScoreError(true);
            }

            if (Number.isNaN(blitzInt)) {
              setBlitzScoreError(true);
            }

            if (!Number.isNaN(dutchInt) && !Number.isNaN(blitzInt)) {
              submitScore(
                Number.parseInt(dutchScore, 10),
                Number.parseInt(blitzScore, 10)
              );
              setDutchScore(0);
              setBlitzScore(10);
            }
          }}
        >
          {editMode ? "Edit Score" : "Submit"}
        </button>
      </div>
    </div>
  );
};
