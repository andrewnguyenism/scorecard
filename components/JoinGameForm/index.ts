export { JoinGameFormError, JoinGameForm } from './JoinGameForm'
