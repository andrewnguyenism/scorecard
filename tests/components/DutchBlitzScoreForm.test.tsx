import user from '@testing-library/user-event'
import { render, screen } from '../test-utils'

import { DutchBlitzScoreForm } from '@/components/DutchBlitzScoreForm'

describe('DutchBlitzScoreForm component', () => {
  it('renders', () => {
    const { asFragment } = render(
      <DutchBlitzScoreForm />
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders in edit mode', () => {
    const { asFragment } = render(
      <DutchBlitzScoreForm editMode />
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('calls submitScore when user presses "Submit"', () => {
    const submitScore = jest.fn()
    const dutchScore = 15
    const blitzScore = 5

    render(
      <DutchBlitzScoreForm submitScore={submitScore} />
    )

    user.type(screen.getByLabelText(/dutch score/i), `${dutchScore}`)
    user.clear(screen.getByLabelText(/blitz score/i))
    user.type(screen.getByLabelText(/blitz score/i), `${blitzScore}`)
    user.click(screen.getByText(/submit/i))

    expect(submitScore).toBeCalledWith(dutchScore, blitzScore)
  })
})
