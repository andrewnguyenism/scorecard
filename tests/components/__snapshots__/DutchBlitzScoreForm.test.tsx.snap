// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DutchBlitzScoreForm component renders 1`] = `
<DocumentFragment>
  <div
    class="rounded-2xl border-2 overflow-hidden m-4 p-6"
  >
    <div
      class="flex flex-col mb-4"
    >
      <label
        class="uppercase font-bold text-gray-900"
        for="dutch"
      >
        Dutch Score
      </label>
      <input
        class="border py-2 px-3 text-grey-darkest false"
        id="dutch"
        inputmode="numeric"
        pattern="[0-9]*"
        placeholder="Dutch score"
        type="text"
        value="0"
      />
    </div>
    <div
      class="flex flex-col mb-4"
    >
      <label
        class="uppercase font-bold text-gray-900"
        for="blitz"
      >
        Blitz Score
      </label>
      <input
        class="border py-2 px-3 text-grey-darkest false"
        id="blitz"
        inputmode="numeric"
        pattern="[0-9]*"
        placeholder="Blitz score"
        type="text"
        value="10"
      />
    </div>
    <div
      class="flex flex-col mb-4 text-center"
    >
      <div
        class="uppercase font-bold text-gray-900"
      >
        Score
      </div>
      <div
        class="text-lg font-bold"
      >
        -20
      </div>
    </div>
    <div>
      <button
        class="block bg-indigo-800 hover:bg-indigo-700 text-white uppercase text-sm font-semibold mx-auto px-4 py-2 rounded-2xl"
      >
        Submit
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`DutchBlitzScoreForm component renders in edit mode 1`] = `
<DocumentFragment>
  <div
    class="rounded-2xl border-2 overflow-hidden m-4 p-6"
  >
    <div
      class="flex flex-col mb-4"
    >
      <label
        class="uppercase font-bold text-gray-900"
        for="dutch"
      >
        Dutch Score
      </label>
      <input
        class="border py-2 px-3 text-grey-darkest false"
        id="dutch"
        inputmode="numeric"
        pattern="[0-9]*"
        placeholder="Dutch score"
        type="text"
        value="0"
      />
    </div>
    <div
      class="flex flex-col mb-4"
    >
      <label
        class="uppercase font-bold text-gray-900"
        for="blitz"
      >
        Blitz Score
      </label>
      <input
        class="border py-2 px-3 text-grey-darkest false"
        id="blitz"
        inputmode="numeric"
        pattern="[0-9]*"
        placeholder="Blitz score"
        type="text"
        value="10"
      />
    </div>
    <div
      class="flex flex-col mb-4 text-center"
    >
      <div
        class="uppercase font-bold text-gray-900"
      >
        Score
      </div>
      <div
        class="text-lg font-bold"
      >
        -20
      </div>
    </div>
    <div>
      <button
        class="block bg-indigo-800 hover:bg-indigo-700 text-white uppercase text-sm font-semibold mx-auto px-4 py-2 rounded-2xl"
      >
        Edit Score
      </button>
    </div>
  </div>
</DocumentFragment>
`;
